main.go

```diff
@@ -0,0 +1,310 @@
+	mysqlDatabase string = ""
+	riemannHost   string = "localhost"
+	riemannPort   string = "5555"
+	riemannTTL    float32
+	riemannTags   []string
+
+	hostname   string
+	interval   time.Duration = time.Second * 30
+	delay      float64       = 2.0
+	log        log15.Logger
+	configFile string
+	debug      bool
+)
+
+func init() {
+	var (
+		h   log15.Handler
+		err error
+	)
+
+	flag.BoolVar(&debug, "d", false, "run in debug mode")
+	flag.StringVar(&configFile, "f", "/etc/riemann-mysql.conf", "path to configuration file")
+	flag.Parse()
+
+	log = log15.New()
+	if debug {
+		h = log15.LvlFilterHandler(log15.LvlDebug, log15.StderrHandler)
+	} else {
+		if h, err = log15.SyslogHandler(syslog.LOG_INFO|syslog.LOG_LOCAL0, "riemann-mysql",
+			log15.LogfmtFormat()); err != nil {
+			fmt.Fprintf(os.Stderr, "error: unable to initialize syslog logging: %s", err)
+			os.Exit(1)
+		}
+		h = log15.LvlFilterHandler(log15.LvlInfo, h)
+	}
+	log.SetHandler(h)
+
+	if configFile != "" {
+		log.Debug("loading configuratin file", "path", configFile)
+		if err := loadConfig(configFile); err != nil {
+			dieOnError(fmt.Sprintf("unable to load configuration: %s", err))
+		}
+	}
+
+	riemannTTL = float32(interval + time.Duration(delay))
+}
+
+func loadConfig(path string) error {
+	var (
+		k, v string
+	)
+
+	f, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if len(line) == 0 || strings.HasPrefix(strings.TrimSpace(line), "#") {
+			continue
+		}
+
+		items := strings.Split(line, "=")
+		if len(items) != 2 {
+			return fmt.Errorf("malformated line %q", line)
+		}
+
+		k, v = strings.TrimSpace(items[0]), strings.TrimSpace(items[1])
+		log.Debug("parsed configuration line",
+			"key", k,
+			"value", v)
+
+		switch k {
+		case "mysql_host":
+			mysqlHost = v
+
+		case "mysql_port":
+			mysqlPort = v
+
+		case "mysql_user":
+			mysqlUser = v
+
+		case "mysql_password":
+			mysqlPassword = v
+
+		case "mysql_database":
+			mysqlDatabase = v
+
+		case "riemann_host":
+			riemannHost = v
+
+		case "riemann_port":
+			riemannPort = v
+
+		case "interval":
+			i, err := strconv.ParseInt(v, 10, 32)
+			if err != nil {
+				return fmt.Errorf("invalid value %q for setting `interval`", v)
+			}
+			interval = time.Duration(i) * time.Second
+
+		case "delay":
+			d, err := strconv.ParseFloat(v, 32)
+			if err != nil {
+				return fmt.Errorf("invalid value %q for setting `delay`", v)
+			}
+			delay = d
+
+		case "hostname":
+			hostname = v
+
+		case "tags":
+			riemannTags = strings.Split(v, " ")
+
+		default:
+			log.Warn(fmt.Sprintf("unsupported configuration setting %q", k))
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func main() {
+	var (
+		riemann *raidman.Client
+		db      *mysql.Conn
+		err     error
+	)
+
+	// Handle termination signals
+	sig := make(chan os.Signal, 1)
+	stop := make(chan bool, 1)
```

Comment:
You could use tomb instead of a stop chan.

Reply:
Done.